Sources/NIOSSH/SSHPacketSerializer.swift

```diff
@@ -47,40 +47,61 @@ struct SSHPacketSerializer {
                 preconditionFailure("only .version message is allowed at this point")
             }
         case .cleartext:
-            let index = buffer.writerIndex
+            buffer.writeSSHPacket(message: message, lengthEncrypted: true, blockSize: 8)
+            self.sequenceNumber &+= 1
+        case .encrypted(let protection):
+            let index = buffer.readerIndex
```

Comment:
this is the bit I don't like, and haven't come up with a better approach yet, why this fiddling with indices is need is not clear